From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xuyin <1573880184@qq.com>
Date: Sun, 8 Dec 2024 21:32:39 +0800
Subject: [PATCH] Removed Folia supported.


diff --git a/src/main/java/io/papermc/paper/plugin/configuration/PluginMeta.java b/src/main/java/io/papermc/paper/plugin/configuration/PluginMeta.java
index 2dbc9fc268850c85e4e8d38da05586a004862c76..1ca527873eac3278a47c59378b0d43377a7655d5 100644
--- a/src/main/java/io/papermc/paper/plugin/configuration/PluginMeta.java
+++ b/src/main/java/io/papermc/paper/plugin/configuration/PluginMeta.java
@@ -183,12 +183,14 @@ public interface PluginMeta {
      */
     @Nullable String getAPIVersion();
 
-    // Folia start - block plugins not marked as supported
-    /**
-     * Returns whether the plugin has been marked to be compatible with regionised threading as provided
-     * by Folia
-     */
-    public boolean isFoliaSupported();
-    // Folia end - block plugins not marked as supported
+    // Fire start - Removed Folia supported.
+//    // Folia start - block plugins not marked as supported
+//    /**
+//     * Returns whether the plugin has been marked to be compatible with regionised threading as provided
+//     * by Folia
+//     */
+//    public boolean isFoliaSupported();
+//    // Folia end - block plugins not marked as supported
+    // Fire end - Removed Folia supported.
 
 }
diff --git a/src/main/java/org/bukkit/plugin/PluginDescriptionFile.java b/src/main/java/org/bukkit/plugin/PluginDescriptionFile.java
index 90cd9568363a7532d600b627de605c12a7b816c7..8313bd11722b6637b515fb5ac86d0d0e602215ee 100644
--- a/src/main/java/org/bukkit/plugin/PluginDescriptionFile.java
+++ b/src/main/java/org/bukkit/plugin/PluginDescriptionFile.java
@@ -267,19 +267,21 @@ public final class PluginDescriptionFile implements io.papermc.paper.plugin.conf
         return this.paperPluginLoader;
     }
     // Paper end - plugin loader api
-    // Folia start - block plugins not marked as supported
-    private String foliaSupported;
-    private static final String FOLIA_SUPPORTED_KEY = "folia-supported";
-
-    /**
-     * Returns whether the plugin has been marked to be compatible with regionised threading as provided
-     * by Folia
-     */
-    @Override
-    public boolean isFoliaSupported() {
-        return this.foliaSupported != null && this.foliaSupported.equalsIgnoreCase("true");
-    }
-    // Folia end - block plugins not marked as supported
+    // Fire start - Removed Folia supported.
+//    // Folia start - block plugins not marked as supported
+//    private String foliaSupported;
+//    private static final String FOLIA_SUPPORTED_KEY = "folia-supported";
+//
+//    /**
+//     * Returns whether the plugin has been marked to be compatible with regionised threading as provided
+//     * by Folia
+//     */
+//    @Override
+//    public boolean isFoliaSupported() {
+//        return this.foliaSupported != null && this.foliaSupported.equalsIgnoreCase("true");
+//    }
+//    // Folia end - block plugins not marked as supported
+    // Fire end - Removed Folia supported.
     // Paper start - oh my goddddd
     /**
      * Don't use this.
@@ -1280,11 +1282,13 @@ public final class PluginDescriptionFile implements io.papermc.paper.plugin.conf
         if (map.get("prefix") != null) {
             prefix = map.get("prefix").toString();
         }
-        // Folia start - block plugins not marked as supported
-        if (map.get(FOLIA_SUPPORTED_KEY) != null) {
-            foliaSupported = map.get(FOLIA_SUPPORTED_KEY).toString();
-        }
-        // Folia end - block plugins not marked as supported
+        // Fire start - Removed Folia supported.
+//        // Folia start - block plugins not marked as supported
+//        if (map.get(FOLIA_SUPPORTED_KEY) != null) {
+//            foliaSupported = map.get(FOLIA_SUPPORTED_KEY).toString();
+//        }
+//        // Folia end - block plugins not marked as supported
+        // Fire end - Removed Folia supported.
     }
 
     @NotNull
@@ -1361,11 +1365,13 @@ public final class PluginDescriptionFile implements io.papermc.paper.plugin.conf
         if (prefix != null) {
             map.put("prefix", prefix);
         }
-        // Folia start - block plugins not marked as supported
-        if (foliaSupported != null) {
-            map.put(FOLIA_SUPPORTED_KEY, foliaSupported);
-        }
-        // Folia end - block plugins not marked as supported
+        // Fire start - Removed Folia supported.
+//        // Folia start - block plugins not marked as supported
+//        if (foliaSupported != null) {
+//            map.put(FOLIA_SUPPORTED_KEY, foliaSupported);
+//        }
+//        // Folia end - block plugins not marked as supported
+        // Fire end - Removed Folia supported.
 
         return map;
     }
